src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,302 @@
+                raise ValueError(f"Expected `feature_tokens` shape to be (n, ) but found {feature_tokens.shape}")
+        if audio is not None:
+            if len(audio.shape) != 1:
+                raise ValueError(f"Expected `audio` shape to be (n, ) but found {feature_tokens.shape}")
+        n_bars = self.n_bars if n_bars is None else n_bars
+
+        if beatstep[0] > 0.01:
+            warnings.warn(f"Inference Warning : beatstep[0] is not 0 ({beatstep[0]}). all beatstep will be shifted.")
+            beatstep = beatstep - beatstep[0]
+
+        if self.use_mel:
+            batch, ext_beatstep = self.preprocess_mel(
+                audio,
+                beatstep,
+                n_bars=n_bars,
+                padding_value=self.padding_value,
+            )
+        else:
+            raise NotImplementedError("use_mel must be True")
+
+        return batch, ext_beatstep
+
+    def __call__(
+        self,
+        raw_audio: Union[np.ndarray, List[float], List[np.ndarray]],
+        audio_sr: int,
+        steps_per_beat: int = 2,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        Args:
+        Main method to featurize and prepare for the model one sequence. Please note that `Pop2PianoFeatureExtractor`
+        only accepts one raw_audio at a time.
+            raw_audio (`np.ndarray`, `List`):
+                Denotes the raw_audio.
+            audio_sr (`int`):
+                Denotes the Sampling Rate of `raw_audio`.
+            steps_per_beat (`int`, *optional*, defaults to 2):
+                Denotes Steps per beat.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*, defaults to 'pt'):
+                If set, will return tensors instead of list of python integers. Acceptable values are:
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return Numpy `np.ndarray` objects.
+        """
+        warnings.warn("Please make sure to have the audio sampling_rate as 44100, to get the optimal performence!")
+        warnings.warn(
+            "Pop2PianoFeatureExtractor only takes one raw_audio at a time, if you want to extract features from more than a single audio then you might need to call it multiple times."
+        )
+
+        # If it's [np.ndarray]
+        if isinstance(raw_audio, list) and isinstance(raw_audio[0], np.ndarray):
+            raw_audio = raw_audio[0]
+
+        bpm, beat_times, confidence, estimates, essentia_beat_intervals = self.extract_rhythm(raw_audio=raw_audio)
+        beatsteps = self.interpolate_beat_times(beat_times, steps_per_beat, extend=True)
+
+        if self.sampling_rate != audio_sr and self.sampling_rate is not None:
+            # Change audio_sr to self.sampling_rate
+            raw_audio = librosa.core.resample(
+                raw_audio, orig_sr=audio_sr, target_sr=self.sampling_rate, res_type="kaiser_best"
+            )
```

Comment:
Hi @ArthurZucker I tried to remove the resampling part but here raw_audio is used 2 times first time in `extract_rhythm` which takes audio with original sampling_rate and the second time in `single_preprocess` which first upscales/downscales raw_audio to sampling_rate of 22050 and then it is passed to the model. Since two parts  of the feature extractor uses the audio at different sampling_rates we need to sample at least one here. 
One thing we could do is we can take the raw_audio at two different sampling_rates but that will be a overhead to the user!